scripts/deploy-erc1155-non-transferable.ts

```diff
@@ -0,0 +1,113 @@
+import hre from 'hardhat';
+import fs from 'fs';
+import {isValidEnv, addressesDirPath, getAddressesFilePath} from './utils';
+import packageFile from '../package.json';
+
+const ethers = hre.ethers;
+
+/**
+ * Class DeploymentExecutor.
+ *
+ * @class DeploymentExecutor
+ */
+class DeploymentExecutor {
+  env;
+  erc1155NonTransferable;
+  maxTip;
+  txOptions;
+
+  constructor() {
+    this.env;
+    this.erc1155NonTransferable;
+
+    this.maxTip = ethers.utils.parseUnits(
+      process.env.MAX_TIP ? String(process.env.MAX_TIP) : '1',
+      'gwei'
+    );
+
+    this.txOptions = {
+      maxPriorityFeePerGas: this.maxTip,
+      gasLimit: 4500000, // our current max contract is around 4.2m gas
+    };
+  }
+
+  async deployContracts() {
+    const signers = await ethers.getSigners();
+    let [, ccTokenDeployer] = signers;
+    const [primaryDeployer] = signers;
```

Comment:
I don't think this script needs the primaryDeployer variable.

Reply:
It was easier (=lazier) to keep it in, since it handeled properly both cases when cc and primary are the same or different.

But I updated it now to properly reflect what's going on.